docs/explanations/rendering.md

```diff
@@ -0,0 +1,344 @@
+
+![example-1320](images/example-1320.png)
+
+# Example: #845
+
+In #845 we are drawing a multi-scale image which is chunked on disk.
+
+## Chunk Size
+
+It's confusing but there can be different chunk sizes in use at one time. If
+using **Dask** it's chunks are often larger than the file format's chunks. This
+means loading one Dask chunk can cause many disk chunks to load into memory. We
+might choose our rendering chunks to be the same size as Dask is using, if we
+can even determine that, or we might chose a different size.
+
+In the end there are two different types of speed: framerate and load time.
+Sometimes there is a tradeoff and we can speed up loading by slowing the
+framerate a bit. Hopefully we can come up with defaults that work well for most
+people, but we'll probably need to provide a way for the user to tune the chunks
+if necessary.
+
+## Octree
+
+To solve #1320 our chunks will be layers. The ChunkManager can write the data
+into the `Image` object for those layers. However with #845 chunks are spatial
+so we need a new spatial datastructure that can keep track of which chunks are
+in memory and store the per-chunk data.
+
+We are doing to use an octree. See
+[Apple's](https://developer.apple.com/documentation/gameplaykit/gkoctree) nice
+illustration of an octree:
+
+![octree](images/octree.png)
+
+In a quadtree every square node is divided into 4 children representing the 4
+spatial quadrants of that node: upper-left, upper-right, lower-left and
+lower-right. An octree is the same idea but in 3D: every node has up to 8 children,
+the 4 on top and the 4 on the bottom.
+
+We can use our octree for 2D situations just by restricting ourselves to the top
+4 children. So we plan to always use the same octree datastructure all the time,
+but use it in 2D or 3D modes as needed.
+
+## Multi-resolution
+
+Like image pyramids the octree can store many versions of the same data at
+different resolutions. The root node contains a downsampled depiction of the
+entire dataset. As the user zooms in, we descend into child nodes which contain
+ever smaller portions of the data, but at ever higher resolutions.
+
+In either case if a chunk is not in memory it will be requested from the
+`ChunkManager`. Until the data is in memory the renderer will draw a
+placeholder. In many cases the best placeholder will be from a different level
+of the octree. This will produce the often seen effect where the image is
+initially blurry but then "refines" as more data is loaded. In the worst case if
+no stand-in is available the placeholder can be blank or a grid potentially with
+a "loading" animation.
+
+## Beyond Images
+
+We are starting with 2D images but we are going to build the `ChunkManager` and
+octree in a generic way so that we can add in more layer types as we go,
+including 3D images, points, shapes and meshes. 2D images are the simplest case,
+but we believe most the infrastructure can be used by the they other layers
+times.
```

Comment:
?

Reply:
Fixed to say `can be used by the other layer types`